### PR TITLE
Use latest version of GraalVM, update deps

### PR DESCRIPTION
**Pull request by burinc**
Changes logs

 - Update version of GraalVM to 21.0.0
 - Bump version of dependencies in project.clj and version.txt
 - Update github script
 - Update docke Dockerfile

### CODE REVIEW

**Comment by burinc** [comment]
I test build locally without any issue:

```
\cat doc/*.adoc > resources/docs.adoc
env LEIN_SNAPSHOTS_IN_RELEASE=1 lein uberjar
Compiling dad.config
Compiling dad.core
Compiling dad.logger
Compiling dad.nrepl
Compiling dad.os
Compiling dad.reader
Compiling dad.reader.impl
Compiling dad.repl
Compiling dad.runner
Compiling dad.runner.impl
Compiling dad.util
Created /home/b12n/codes/dad/target/dad-0.3.2.jar
Created /home/b12n/codes/dad/target/dad.jar
/usr/lib/jvm/java-11-graalvm/bin/native-image \
	-jar target/dad.jar \
	-H:Name=dad \
	-H:+ReportExceptionStackTraces \
	-J-Dclojure.spec.skip-macros=true \
	-J-Dclojure.compiler.direct-linking=true \
	"-H:IncludeResources=command.edn" \
	"-H:IncludeResources=schema.edn" \
	"-H:IncludeResources=config.edn" \
	"-H:IncludeResources=version.txt" \
	"-H:IncludeResources=docs.adoc" \
	--initialize-at-build-time  \
	--report-unsupported-elements-at-runtime \
	-H:Log=registerResource: \
	--verbose \
	--no-fallback \
	--no-server \
	"--static" \
	"-J-Xmx3g"
Executing [
/usr/lib/jvm/java-11-graalvm/bin/java \
-XX:+UseParallelGC \
-XX:+UnlockExperimentalVMOptions \
-XX:+EnableJVMCI \
-Dtruffle.TrustAllTruffleRuntimeProviders=true \
-Dtruffle.TruffleRuntime=com.oracle.truffle.api.impl.DefaultTruffleRuntime \
-Dgraalvm.ForcePolyglotInvalid=true \
-Dgraalvm.locatorDisabled=true \
-Dsubstratevm.IgnoreGraalVersionCheck=true \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.aarch64=ALL-UNNAMED \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.amd64=ALL-UNNAMED \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.code.site=ALL-UNNAMED \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.code.stack=ALL-UNNAMED \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.code=ALL-UNNAMED \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.common=ALL-UNNAMED \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.hotspot.aarch64=ALL-UNNAMED \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.hotspot.amd64=ALL-UNNAMED \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.hotspot.sparc=ALL-UNNAMED \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.hotspot=ALL-UNNAMED \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.meta=ALL-UNNAMED \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.runtime=ALL-UNNAMED \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.services=ALL-UNNAMED \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.sparc=ALL-UNNAMED \
--add-exports=org.graalvm.truffle/com.oracle.truffle.api=ALL-UNNAMED \
--add-opens=jdk.internal.vm.compiler/org.graalvm.compiler.debug=ALL-UNNAMED \
--add-opens=jdk.internal.vm.compiler/org.graalvm.compiler.nodes=ALL-UNNAMED \
--add-opens=jdk.unsupported/sun.reflect=ALL-UNNAMED \
--add-opens=java.base/jdk.internal.module=ALL-UNNAMED \
--add-opens=java.base/jdk.internal.ref=ALL-UNNAMED \
--add-opens=java.base/jdk.internal.reflect=ALL-UNNAMED \
--add-opens=java.base/java.io=ALL-UNNAMED \
--add-opens=java.base/java.lang=ALL-UNNAMED \
--add-opens=java.base/java.lang.reflect=ALL-UNNAMED \
--add-opens=java.base/java.lang.invoke=ALL-UNNAMED \
--add-opens=java.base/java.lang.ref=ALL-UNNAMED \
--add-opens=java.base/java.net=ALL-UNNAMED \
--add-opens=java.base/java.nio=ALL-UNNAMED \
--add-opens=java.base/java.nio.file=ALL-UNNAMED \
--add-opens=java.base/java.security=ALL-UNNAMED \
--add-opens=java.base/javax.crypto=ALL-UNNAMED \
--add-opens=java.base/java.util=ALL-UNNAMED \
--add-opens=java.base/java.util.concurrent.atomic=ALL-UNNAMED \
--add-opens=java.base/sun.security.x509=ALL-UNNAMED \
--add-opens=java.base/jdk.internal.logger=ALL-UNNAMED \
--add-opens=org.graalvm.sdk/org.graalvm.nativeimage.impl=ALL-UNNAMED \
--add-opens=org.graalvm.sdk/org.graalvm.polyglot=ALL-UNNAMED \
--add-opens=org.graalvm.truffle/com.oracle.truffle.polyglot=ALL-UNNAMED \
--add-opens=org.graalvm.truffle/com.oracle.truffle.api.impl=ALL-UNNAMED \
-XX:+UseJVMCINativeLibrary \
-Xss10m \
-Xms1g \
-Xmx14g \
-Duser.country=US \
-Duser.language=en \
-Djava.awt.headless=true \
-Dorg.graalvm.version=21.0.0 \
-Dorg.graalvm.config= \
-Dcom.oracle.graalvm.isaot=true \
-Djava.system.class.loader=com.oracle.svm.hosted.NativeImageSystemClassLoader \
-Dawt.toolkit=sun.awt.X11.XToolkit \
-Djava.awt.graphicsenv=sun.awt.X11GraphicsEnvironment \
-Djava.awt.printerjob=sun.print.PSPrinterJob \
-Xshare:off \
--module-path \
/usr/lib/jvm/java-11-graalvm/lib/truffle/truffle-api.jar \
-Djdk.internal.lambda.disableEagerInitialization=true \
-Djdk.internal.lambda.eagerlyInitialize=false \
-Djava.lang.invoke.InnerClassLambdaMetafactory.initializeLambdas=false \
-Dclojure.spec.skip-macros=true \
-Dclojure.compiler.direct-linking=true \
-Xmx3g \
-javaagent:/usr/lib/jvm/java-11-graalvm/lib/svm/builder/svm.jar \
-cp \
/usr/lib/jvm/java-11-graalvm/lib/svm/builder/pointsto.jar:/usr/lib/jvm/java-11-graalvm/lib/svm/builder/javacpp-shadowed.jar:/usr/lib/jvm/java-11-graalvm/lib/svm/builder/llvm-platform-specific-shadowed.jar:/usr/lib/jvm/java-11-graalvm/lib/svm/builder/svm-llvm.jar:/usr/lib/jvm/java-11-graalvm/lib/svm/builder/llvm-wrapper-shadowed.jar:/usr/lib/jvm/java-11-graalvm/lib/svm/builder/svm.jar:/usr/lib/jvm/java-11-graalvm/lib/svm/builder/objectfile.jar \
'com.oracle.svm.hosted.NativeImageGeneratorRunner$JDK9Plus' \
-watchpid \
103566 \
-imagecp \
/usr/lib/jvm/java-11-graalvm/lib/svm/library-support.jar:/home/b12n/codes/dad/target/dad.jar \
-H:Path=/home/b12n/codes/dad \
-H:Class=dad.core \
-H:Name=dad \
-H:+ReportExceptionStackTraces \
-H:IncludeResources=command.edn \
-H:IncludeResources=schema.edn \
-H:IncludeResources=config.edn \
-H:IncludeResources=version.txt \
-H:IncludeResources=docs.adoc \
-H:ClassInitialization=:build_time \
-H:+ReportUnsupportedElementsAtRuntime \
-H:Log=registerResource: \
-H:FallbackThreshold=0 \
-H:+StaticExecutable \
-H:CLibraryPath=/usr/lib/jvm/java-11-graalvm/lib/svm/clibraries/linux-amd64 \

]
[dad:103586]    classlist:     898.50 ms,  0.96 GB
[dad:103586]        (cap):     412.85 ms,  0.96 GB
[dad:103586]        setup:   1,491.38 ms,  0.96 GB
[Use -Dgraal.LogFile=<path> to redirect Graal log output to a file.]
[thread:28] scope: ForkJoinPool-4-worker-19
  [thread:28] scope: ForkJoinPool-4-worker-19.registerResource
  ServiceLoaderFeature: registerResource: META-INF/services/java.nio.file.spi.FileSystemProvider
  [thread:28] scope: ForkJoinPool-4-worker-19.registerResource
  Resources have been added by ServiceLoaderFeature. Automatic registration can be disabled with -H:-UseServiceLoaderFeature
[dad:103586]     (clinit):     304.72 ms,  1.66 GB
[dad:103586]   (typeflow):   8,817.42 ms,  1.66 GB
[dad:103586]    (objects):  12,662.74 ms,  1.66 GB
[dad:103586]   (features):     614.05 ms,  1.66 GB
[dad:103586]     analysis:  23,940.55 ms,  1.66 GB
[dad:103586]     universe:     855.44 ms,  1.66 GB
[dad:103586]      (parse):   2,627.79 ms,  1.99 GB
[dad:103586]     (inline):   3,206.84 ms,  2.39 GB
[dad:103586]    (compile):  16,101.71 ms,  2.48 GB
[dad:103586]      compile:  23,146.15 ms,  2.48 GB
[dad:103586]        image:   3,423.04 ms,  2.48 GB
[dad:103586]        write:     432.20 ms,  2.48 GB
[dad:103586]      [total]:  54,351.29 ms,  2.48 GB
```

Test seems to be failing for some reason.

**Comment by liquidz** [comment]
@burinc Wow, great work! Thanks!
I'll have a look.

File: CHANGELOG.adoc [hunk 1]
```diff
@@ -4,6 +4,12 @@ All notable changes to this project will be documented in this file. This change
 
 == Unreleased (dev)
 
+== 0.3.2 (2020-01-19)
+// {{{
+=== Changed
+- https://github.com/liquidz/dad/issues/13[#13]: Bump GraalVM to 21.0.0.
```
Comment (liquidz) [inline CHANGELOG.adoc]: Is this link correct?

**Review by liquidz — COMMENTED** [review]
seems `./script/linux-build` is failing.
it maybe the problem of docker image, so I'll test more.

**Comment by liquidz** [comment]
It seems `ubuntu:latest` does not contain `curl`.
Could you add following diffs? @burinc 

```diff
diff --git a/test/resources/test_task/pre_test b/test/resources/test_task/pre_test
index 228372c..126cf79 100644
--- a/test/resources/test_task/pre_test
+++ b/test/resources/test_task/pre_test
@@ -14,9 +14,9 @@ if [ ! -e ${BASE_DIR}/pre/dummy ]; then
     exit 1
 fi
 
-${DAD_CMD} --eval '(package ["git" "cowsay"])'
+${DAD_CMD} --eval '(package ["git" "cowsay" "curl"])'
 if [ $? -ne 0 ]; then
-    echo "pre: failed to pre-install git or cowsay."
+    echo "pre: failed to pre-install git, cowsay or curl."
     exit 1
 fi
```

File: CHANGELOG.adoc (hunk 1 above)
Comment (burinc) [reply CHANGELOG.adoc]: You are right. fixed.

**Comment by burinc** [comment]
> It seems `ubuntu:latest` does not contain `curl`.
> Could you add following diffs? @burinc
> 
> ```diff
> diff --git a/test/resources/test_task/pre_test b/test/resources/test_task/pre_test
> index 228372c..126cf79 100644
> --- a/test/resources/test_task/pre_test
> +++ b/test/resources/test_task/pre_test
> @@ -14,9 +14,9 @@ if [ ! -e ${BASE_DIR}/pre/dummy ]; then
>      exit 1
>  fi
>  
> -${DAD_CMD} --eval '(package ["git" "cowsay"])'
> +${DAD_CMD} --eval '(package ["git" "cowsay" "curl"])'
>  if [ $? -ne 0 ]; then
> -    echo "pre: failed to pre-install git or cowsay."
> +    echo "pre: failed to pre-install git, cowsay or curl."
>      exit 1
>  fi
> ```

Fixed in the last commit.

**Comment by liquidz** [comment]
Thanks!
I'll cut a new release later.

**Pull request merged**